### PR TITLE
Update StatusCommandParserTestCase

### PR DESCRIPTION
**Pull request by woke02**
## Description
Closes [#207](https://github.com/AY2425S1-CS2103T-W09-3/tp/issues/207)

This PR enhances the `StatusCommand` class by addressing [brief summary of the problem]. Specifically, it [short explanation of the fix or feature]. This change ensures that [benefit or improvement introduced].

### Before
- Previous behavior did not handle [specific issue or limitation, e.g., null statuses or unsupported values].

### After
- Added checks to handle cases where `Status` is null or unsupported, ensuring that a `ParseException` is thrown with appropriate messages.
- Enhanced test coverage to include edge cases, such as non-numeric indices, negative indices, and additional whitespace.

| Scenario                    | Previous Behavior                | Current Behavior                 |
|-----------------------------|-----------------------------------|-----------------------------------|
| Null `Status`               | Unhandled                         | Throws `ParseException`           |
| Unsupported `Status`        | Unhandled                         | Throws `ParseException`           |
| Non-numeric index input     | ParseException (inconsistent)     | Consistently throws `ParseException` with a user-friendly message |

## Impact of Changes
- Improved error handling for `StatusCommand` to cover edge cases.
- Enhanced user experience through better feedback when incorrect input is provided.
- Ensured consistent behavior across different input types, reducing unexpected failures.

## Testing Steps
1. Navigate to the [specific feature] where `StatusCommand` is utilized.
2. Input test cases:
   - Valid indices with valid statuses.
   - Indices with null or unsupported statuses.
   - Edge cases such as negative and zero indices.
3. Verify that a `ParseException` is raised for invalid cases and that valid cases proceed as expected.

### Automated Tests
- Added new unit tests in `StatusCommandParserTest` to cover scenarios:
  - Null and unsupported `Status` handling.
  - Non-numeric and special character index handling.
  - Edge cases for zero and negative index input.

### CODE REVIEW

File: src/test/java/seedu/hireme/logic/commands/StatusCommandTest.java [hunk 1]
```diff
@@ -116,25 +116,41 @@ public void equals() {
...
         StatusCommand statusDifferentStatusCommand =
                 new StatusCommand(INDEX_FIRST_INTERNSHIP_APPLICATION, Status.PENDING);
-        assertFalse(statusFirstCommand.equals(statusDifferentStatusCommand));
+        assertFalse(statusFirstCommand.equals(statusDifferentStatusCommand),
+                "Objects with the same index but different statuses should not be equal");
+
+        // Test: Objects with different indices and different statuses should not be equal
+        StatusCommand completelyDifferentCommand =
+                new StatusCommand(INDEX_SECOND_INTERNSHIP_APPLICATION, Status.REJECTED);
+        assertFalse(statusFirstCommand.equals(completelyDifferentCommand),
```
Comment (snowstopxt) [inline src/test/java/seedu/hireme/logic/commands/StatusCommandTest.java:146]: funny name

**Review by snowstopxt — APPROVED** [review]
LGTM

**Review by Raghava-Chittidi — APPROVED** [review]
LGTM!
